pkg/nad/masterplugin.go

```diff
@@ -393,3 +393,66 @@ func (plugin *MasterIPVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error)
 
 	return plugin.masterPlugin, nil
 }
+
+// MasterBoundPlugin provides struct for MasterPlugin to create a NeNetworkAttachmentDefinition Master plugin for
+// bonded interfaces.
+type MasterBoundPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterBoundPlugin define network-attachment-interface for bond interface.
+func NewMasterBoundPlugin(name, mode string) *MasterBoundPlugin {
+	glog.V(100).Infof(
+		"Initializing new MasterBoundPlugin structure %s", name)
+
+	builder := MasterBoundPlugin{
+		masterPlugin: &MasterPlugin{
+			CniVersion:      "0.3.1",
+			Name:            name,
+			Type:            "bond",
+			Mode:            mode,
+			FailOverMac:     1,
+			LinkInContainer: true,
+			Miimon:          "100",
+			Links:           []Link{{Name: "net1"}, {Name: "net2"}},
+			Capabilities:    &Capability{IP: true},
+		},
+	}
+
+	if builder.masterPlugin.Name == "" {
+		glog.V(100).Infof("error NewMasterBoundPlugin name can not be empty")
+
+		builder.errorMsg = "MasterMacVlanPlugin name is empty"
+	}
+
+	if builder.masterPlugin.Mode == "" {
+		glog.V(100).Infof("error NewMasterBoundPlugin Mode can not be empty")
+
+		builder.errorMsg = "NewMasterBoundPlugin Mode is empty"
+	}
+
+	return &builder
+}
+
+// WithMTU defines nad with MTU.
+func (plugin *MasterBoundPlugin) WithMTU(mtu string) *MasterBoundPlugin {
+	if mtu == "" {
```

Comment:
You need to validate that mtu in the supported MTU range 1-9192. For that you need to change MTU parameter type to int and chack that this value in range:
func (plugin *MasterBoundPlugin) WithMTU(mtu int) *MasterBoundPlugin {
...
	if 1 > mtu || mtu > 9192 {
		builder.errorMsg = fmt.Sprintf("invalid mtu size %d allowed mtu should be in range 1...9192", mtu)
	}

Reply:
Changed